hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/CallContext.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi;
+
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/** The {@code CallContext} provides access to all services without leaking the underlying state. */
+public interface CallContext {
```

Comment:
Not a fan of the name, I just didn't have anything better on-hand during the zoom call :-). Is this used only during pre-handle? Or is it used during the other workflows as well? Any better ideas for names?

Reply:
Agree, the name should be changed. I chose it, so people recognize the responsibility of this class. 🙂

Currently I do not have a better name though.